app/role.go

```diff
@@ -125,6 +125,8 @@ func (a *App) PatchRole(role *model.Role, patch *model.RolePatch) (*model.Role,
 		return nil, err
 	}
 
+	a.sendUpdatedRoleEvent(role)
```

Comment:
a.UpdateRole is also _sometimes_ sending a role updated event... can we have that function skip the current role on line 212? basically add a condition here
```
	for _, ir := range impactedRoles {
		a.sendUpdatedRoleEvent(ir)
	}
``` 

changed to 

```
	for _, ir := range impactedRoles {
            if ir.Name != role.Name {
		a.sendUpdatedRoleEvent(ir)
            }
	}
```

Also can we change line 172 to be 
`builtInRolesMinusChannelRoles := append(utils.RemoveStringsFromSlice(model.BuiltInSchemeManagedRoleIDs, builtInChannelRoles...), model.NewSystemRoleIDs...)`? to ignore all the channel moderation related logic if its a new system role

Reply:
Done!